### PR TITLE
Increase debug logging

### PR DESCRIPTION
**Pull request by voidshard**
Adds extra log lines to make it clearer what is happening (or not)

# Pull Request Template

## Description

Trivial change to add some extra debug level logging into the adapter.

## Type of change

Logging++

## Environment

n/a

### CODE REVIEW

**Comment by voidshard** [comment]
@seanmalloy @vinny-sabatini just flagging you as I believe we'll need a review here. The merges a single commit that adds some log statements :shrug: should be fairly painless to review.

File: main.go [hunk 1]
```diff
@@ -345,6 +350,7 @@ func sendSamples(logger log.Logger, w writer, timeseries []*prompb.TimeSeries) {
 		failedSamples.WithLabelValues(w.Name()).Add(float64(len(timeseries)))
 		writeErrors.Inc()
 	} else {
+		level.Debug(logger).Log("msg", "Sent samples", "num_samples", len(timeseries))
```
Comment (seanmalloy) [inline main.go:353]: This is already provided as a Prometheus metric, see line 355. What is the use case for having is as a log and a metric?

File: main.go [hunk 2]
```diff
@@ -268,9 +270,12 @@ func serve(logger log.Logger, addr string, writers []writer, readers []reader) e
 		duration := time.Since(begin).Seconds()
 		writeProcessingDuration.WithLabelValues(writers[0].Name()).Observe(duration)
 
+		level.Debug(logger).Log("msg", "/write", "duration", duration) //nolint:errcheck
```
Comment (seanmalloy) [inline main.go:273]: Similar comment here. This is already provided as a Prometheus metric, see line 271. Do we really need to provide it as log message and metric?

**Comment by seanmalloy** [comment]
> @voidshard Is there a reason for logging on the server side, when most of these are metrics being sent to prometheus? I understanding logging the request url and method. Also we will need a `//nolint:errcheck` comment on these lines. cc: @seanmalloy If you think more server side logging is ok for Splunk or Elastic Search reasons I think the PR is fine after they fix the lint checker.

My perspective is we do need need to add the two log lines that are already provided by Prometheus metrics. See my two inlines comments.

Other than that just fix the lint errors and we should be able to merge this.

**Comment by voidshard** [comment]
> > @voidshard Is there a reason for logging on the server side, when most of these are metrics being sent to prometheus? I understanding logging the request url and method. Also we will need a `//nolint:errcheck` comment on these lines. cc: @seanmalloy If you think more server side logging is ok for Splunk or Elastic Search reasons I think the PR is fine after they fix the lint checker.
> 
> My perspective is we do need need to add the two log lines that are already provided by Prometheus metrics. See my two inlines comments.
> 
> Other than that just fix the lint errors and we should be able to merge this.

Yeah I'm not sure I agree with the sentiment that just because we have something in prom metrics we therefore can't or shouldn't log it (especially in debug logging). Debug logging implies a user is deliberately enabling this setting & wants to see everything that this application is doing, primarily through said logs.

For the case I encountered; we were sending empty timeseries (long story) and it wasn't obvious via metrics (all 0), logs (none) or errors (also 0) what the issue was. The debug logs revealed that the adapter was getting traffic & that there were no metrics to insert, something that metrics alone didn't show.

**Comment by wdoogz** [comment]
> 

Ahhh okay. I see the reasoning for using the debug logging. @voidshard can you add the `//nolint:errcheck` comment to your changes?
@seanmalloy This will require the user to enable debugging on the adapter using the `--log.level` flag, it should be okay.

**Review by wdoogz — APPROVED** [review]
LGTM.

**Review by seanmalloy — APPROVED** [review]

File: main.go (hunk 1 above)
Comment (seanmalloy) [reply main.go]: This is fine because it is a debug log.

File: main.go (hunk 2 above)
Comment (seanmalloy) [reply main.go:290]: This is fine because it is a debug log.

**Pull request merged**